core/build.gradle

```diff
@@ -42,90 +20,106 @@ dependencies {
+    keepclass "${shadedPackage}.guava.collect.Tables"
+    keepclass "${shadedPackage}.guava.collect.TreeBasedTable"
+    keepclass "${shadedPackage}.guava.collect.TreeMultimap"
+    keepclass "${shadedPackage}.guava.escape.Escapers"
+    keepclass "${shadedPackage}.guava.io.BaseEncoding"
+    keepclass "${shadedPackage}.guava.io.ByteSink"
+    keepclass "${shadedPackage}.guava.io.ByteStreams"
+    keepclass "${shadedPackage}.guava.io.Resources"
+    keepclass "${shadedPackage}.guava.net.UrlEscapers"
+    keepclass "${shadedPackage}.guava.primitives.UnsignedInteger"
+    keepclass "${shadedPackage}.guava.primitives.UnsignedLong"
+    keepclass "${shadedPackage}.guava.reflect.Invokable"
+    keepclass "${shadedPackage}.guava.reflect.Parameter"
+    keepclass "${shadedPackage}.guava.reflect.Reflection"
+    keepclass "${shadedPackage}.guava.reflect.TypeToken"
+    keepclass "${shadedPackage}.guava.util.concurrent.MoreExecutors"
+    keepclass "${shadedPackage}.guava.util.concurrent.UncheckedExecutionException"
+
+    // Reflections (imported by armeria-thrift)
+    keepclass "${shadedPackage}.reflections.Configuration"
+    keepclass "${shadedPackage}.reflections.Reflections"
+    keepclass "${shadedPackage}.reflections.scanners.ResourcesScanner"
+    keepclass "${shadedPackage}.reflections.util.ClasspathHelper"
+    keepclass "${shadedPackage}.reflections.util.ConfigurationBuilder"
+    keepclass "${shadedPackage}.reflections.util.FilterBuilder"
+
+    // FastUtil (imported by armeria-logback)
+    keepclass "${shadedPackage}.fastutil.objects.Object2ObjectMap"
+    keepclass "${shadedPackage}.fastutil.objects.Object2ObjectFunction"
+    keepclass "${shadedPackage}.fastutil.objects.Object2ObjectOpenHashMap"
 }
 
-task relocateInternalDependencies(type: ShadowJar, dependsOn: tasks.trimInternalDependencies) {
-    destinationDir = internalDependenciesJarRelocated.parentFile
-    archiveName = internalDependenciesJarRelocated.name
-    // TODO(trustin): Uncomment this for faster building once shadow 2.0 is out.
-    //                https://github.com/johnrengelman/shadow/issues/203
-    //entryCompression = ZipEntryCompression.STORED
+tasks.assemble.dependsOn tasks.trimShadedJar
 
-    configurations = [ project.configurations.trimmedInternalDependency ]
-
-    relocations.each {
-        relocate it[0], it[1]
+// Add the trimmed JAR to archives so that the 'signArchives' task signs it.
+artifacts {
+    trimShadedJar.outJarFiles.each {
+        archives it
     }
 }
 
-task extractRelocatedDependencies(type: Copy, dependsOn: relocateInternalDependencies) {
-    from(zipTree(internalDependenciesJarRelocated)) {
-        exclude 'META-INF/MANIFEST.MF'
-    }
-    into sourceSets.main.output.classesDir
-}
-jar.dependsOn extractRelocatedDependencies
-
-// Exclude the internal dependencies from the coverage report.
-test {
-    jacoco {
-        excludes += [ '**/internal/deps/**' ]
+// Make sure trimShadedJar task is executed before signing.
+model {
```

Comment:
`signArchives` is created dynamically by the extension, so just attempting to access it will cause a failure.